laser.py

```diff
@@ -130,16 +133,30 @@ def stopLaser( self ):
   def configureScanDataOutput( self ):
     pass
 
+
+def write_timestamp(f):
+  """write time.time() in sec/frac into a binary file"""
+  t = time.time()
+  sec = int(t)
+  frac = int(0x10000*(t-sec))
+  f.write(struct.pack('IH', sec, frac))
+
+
 class LaserIP( Laser ):
   def __init__( self, **kw ):
     self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
     self.socket.connect((HOST, PORT))
+    self.raw_log = open(timeName('logs/laserraw_', 'bin'), 'wb')
+    self.raw_log.write(struct.pack('HH', 0xF472, 2))  # magic number + version 2 with timestamps
+    write_timestamp(self.raw_log)
+    self.raw_log.flush()
     self._buffer = ""
     Laser.__init__( self, **kw )
     self.sendCmd( 'sMN SetAccessMode 03 F4724744' )
 
   def __del__( self ):
     self.socket.close()
+    self.raw_log.close()
```

Comment:
`__del__` is not guaranteed to be called. The only way to be sure `close()` is really called is to use context manager. It might not cause problem in this particular case because you call flush after each write but in general I would not consider this to be a good practice. For the future please consider using context managers for such cases.